src/main/java/org/zalando/nakadi/service/subscription/zk/AbstractZkSubscriptionClient.java

```diff
@@ -257,7 +258,13 @@ public final void unregisterSession(final Session session) {
         return loadDataAsync(
                 zkSessions,
                 key -> getSubscriptionPath("/sessions/" + key),
-                (key, data) -> new Session(key, Integer.parseInt(new String(data, UTF_8)))).values();
+                (key, data) -> {
+                    try {
+                        return deserializeSession(key, data);
+                    } catch (final IOException e) {
+                        throw new NakadiRuntimeException(e);
```

Comment:
why not to put it on the level of deserialisation?

Reply:
I think that deserialization that throws IOException is more logical. So that the callers of that should handle the possible exception.

Reply:
we just throw sand in the wheels by ourselves, I am more than sure will process it the same place in other places, because you can not do much with it having any exception type.